src/controllers/DashboardController.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Cronbeat\Controllers;
+
+use Cronbeat\Views\DashboardView;
+
+class DashboardController extends BaseController {
+    public function doRouting() {
+        // Parse the URL
+        $uri = $_SERVER['REQUEST_URI'];
+        $uri = trim($uri, '/');
+        $uri = explode('/', $uri);
+        
+        // Get the action
+        $action = isset($uri[1]) ? $uri[1] : 'index';
+        
+        // Handle form submissions
+        if ($_SERVER['REQUEST_METHOD'] === 'POST') {
+            $action = 'process';
+        }
+        
+        // Call the appropriate method
+        if (method_exists($this, $action)) {
+            $this->$action();
+        } else {
+            $this->index();
+        }
+    }
```

Comment:
@jetbrains-junie refactor this:

- extract function to parse `path` from `REQUEST_URI` and removes the controller part of it (that was used to identify the controller already) 
- move this function to BaseController
- in the controller implementations then come up with a mapping of string to function, do not use `$this->$action`